app/controllers/concerns/users_controller_template.rb

```diff
@@ -6,16 +6,21 @@ module UsersControllerTemplate
     before_action :set_user!, only: [:show, :update]
     before_action :set_new_user!, only: :create
     before_action :protect_permissions_assignment!, only: [:create, :update]
+    after_action :verify_user_name!, only: [:create, :update]
   end
 
   private
 
+  def verify_user_name!
```

Comment:
What about the following scenario?

1. User Julian creates an account. He enteres `first_name: Dark, last_name: PunPun`
2. He joins to an organization in the real world
3. The organization eventually sets his permissions, through the API. Since this leads to an update, Julian names are verified as _Dark PunPun_, right?
 
Perhaps I am not getting some ideas of the implementation of this feature, but the original intention - or at least, my intention - is to have read-only names and usernames that correspond to legal contexts. Those names should not be editable by user, and by default they should be absent until some admin explicitly sets them. 

This means that this fields will normally be empty, unless when they are entered when the user is created through classroom - because we can assume that teachers verify names before entering them. We could also allow APIs to specify they want to mark names as verified, but I think this should not be a default, implicit operation.